cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -96,6 +99,50 @@ class CwlWorkflowWomSpec extends FlatSpec with Matchers {
     }
   }
 
+  behavior of "A decoded CWL 3step"
+
+  private val stringOrExpressionTests = Table(
+    ("index", "result"),
+    (0, Coproduct[CommandLineTool.StringOrExpression]("grep")),
+    (1, Coproduct[CommandLineTool.StringOrExpression](refineMV[MatchesECMAScriptExpression]("$(inputs.pattern)"))),
+    (2, Coproduct[CommandLineTool.StringOrExpression](refineMV[MatchesECMAScriptExpression]("$(inputs.file)"))),
+    (3, Coproduct[CommandLineTool.StringOrExpression]("|")),
+    (4, Coproduct[CommandLineTool.StringOrExpression]("wc")),
+    (5, Coproduct[CommandLineTool.StringOrExpression]("-l"))
+  )
+
+  private def getTestName(stringOrExpression: CommandLineTool.StringOrExpression): String = {
+    stringOrExpression match {
+      case Inl(expression) => s"expression ${expression.value}"
+      case Inr(Inl(string)) => s"string $string"
```

Comment:
Is this the Right Way to type match with `Coproduct`s?

Reply:
ToL a little unintuitive that `StringOrExpression` puts the expression type first if we're going to be doing ordered matches like this.

Reply:
A Poly1 & Fold is the Right Way ™️